storage/storage_image.go

```diff
@@ -533,19 +537,11 @@ func (s *storageImageDestination) Commit(ctx context.Context) error {
 			lastLayer = layer.ID
 			continue
 		}
-		// Check if we cached a file with that blobsum.  If we didn't already have a layer with
```

Comment:
One last nit, please drop the
```go
var diff io.ReadCloser
```
about 30 lines above, to make it clear that not using the variable outside of the “Try to find the layer with contents matching that blobsum.” block is not an oversight.

LGTM otherwise.

Reply:
Dropped it.  Thanks!